base/random.jl

```diff
@@ -1166,6 +1166,29 @@ immutable UUID
 end
 UUID(u::AbstractString) = convert(UUID, u)
```

Comment:
It would maybe be better to just change the default constructor to `UUID(u::Integer) = new(u)`.